# 07 - Recap: how the list of companies got smaller

Quick recap of the pipeline: I started with 139 companies from the robotics supply chain, and I ended up with 59 tickers in the final PEG vs ROIC chart. This notebook shows where the other companies were dropped.

In [1]:
import pandas as pd
import plotly.express as px

For each step, I just count the rows in the file produced by that step.

In [2]:
stages = {
    "Identified": "../../data/raw/stocks.csv",
    "Public +<br>cleaned": "../../data/processed/stocks_cleaned.csv",
    "Tickers<br>verified": "../../data/processed/stocks_transformed_verified.csv",
    "With KPIs": "../../data/processed/stocks_kpis.csv",
}

counts = {stage: len(pd.read_csv(path)) for stage, path in stages.items()}

# Same filters as the scatter plot
kpis = pd.read_csv(stages["With KPIs"])
in_chart = kpis[(kpis["PEG Ratio (1Y)"] > 0) & (kpis["PEG Ratio (1Y)"] <= 10)]

funnel = pd.DataFrame({
    "stage": [*stages, "In chart"],
    "count": [*counts.values(), len(in_chart)],
})

funnel

,stage,count
0,Identified,139
1,Public +<br>cleaned,105
2,Tickers<br>verified,103
3,With KPIs,71
4,In chart,61


Plot the result as a simple bar chart, one bar per step.

In [3]:
fig = px.bar(
    funnel,
    x="stage",
    y="count",
    text="count",
    color="stage",
    title="Screening funnel",
)

fig.update_traces(textposition="outside", showlegend=False)
fig.update_layout(xaxis_title="", yaxis_title="Companies", showlegend=False)

fig.show()

Details of the cleaning and transformations:

* **139 to 105 (-34):** removed the private companies and the tickers I could not use.
* **105 to 103 (-2):** only 2 tickers could not be found on yfinance.
* **103 to 71 (-32):** for the other tickers, I could not get both the PEG ratio and the ROIC.
* **71 to 61 (-10):** I removed the negative PEG ratios (a negative PEG does not mean cheap) and the outliers above PEG = 10.

Last bar: 61 rows, but only 59 unique tickers, because a few companies appear in two different supply chain levels.